### PR TITLE
🌐 Add Traditional Chinese translation for `docs/zh-hant/docs/async.md`

### PR DESCRIPTION
**Pull request by ILoveSorasakiHina**

### CODE REVIEW

File: docs/zh-hant/docs/async.md [hunk 1]
```diff
@@ -0,0 +1,443 @@
...
+
+### 子依賴項
+
+你可以擁有多個相互依賴的依賴項和[子依賴項](tutorial/dependencies/sub-dependencies.md){.internal-link target=_blank} （作為函數定義的參數），其中一些可能是用 `async def` 宣告，也可能是用 `def` 宣告。它們仍然可以正常運作，用 `def` 定義的那些將會在外部的執行緒中呼叫（來自執行緒池），而不是被「等待」。
+
+### 其他輔助函數
+
+你可以直接呼叫任何使用 `def` 或 `async def` 創建的其他輔助函數，FastAPI 不會影響你呼叫它們的方式。
+
+這與 FastAPI 為你呼叫*路徑操作函數*和依賴項的邏輯有所不同。
```
[the remark this answers is not in the document]
Comment (codingjenny) [reply docs/zh-hant/docs/async.md:435]: 有 * 的 markdown 出現，前後如果是中文的話，要空一格，在網頁上呈現才會是斜體。

File: docs/zh-hant/docs/async.md [hunk 2]
```diff
@@ -0,0 +1,443 @@
...
+
+---
+
+如果你不確定該用哪個，直接用 `def` 就好。
+
+---
+
+**注意**：你可以在路徑操作函數中混合使用 `def` 和 `async def` ，並使用最適合你需求的方式來定義每個函數。FastAPI 會幫你做正確的處理。
+
+無論如何，在上述哪種情況下，FastAPI 仍將以異步方式運行，並且速度非常快。
```
Comment (codingjenny) [inline docs/zh-hant/docs/async.md:54]:
```suggestion
無論如何，在上述哪種情況下，FastAPI 仍將以非同步方式運行，並且速度非常快。
```

File: docs/zh-hant/docs/async.md [hunk 3]
```diff
@@ -0,0 +1,443 @@
...
+
+但透過遵循上述步驟，它將能進行一些效能優化。
+
+## 技術細節
+
+現代版本的 Python 支持使用 **"協程"** 的 **`async` 和 `await`** 語法來寫 **"異步程式碼"**。
```
Comment (codingjenny) [inline docs/zh-hant/docs/async.md:60]:
```suggestion
現代版本的 Python 支援使用 **「協程」** 的 **`async` 和 `await`** 語法來寫 **「非同步程式碼」**。
```
「支持」是簡中用法

File: docs/zh-hant/docs/async.md [hunk 4]
```diff
@@ -0,0 +1,443 @@
...
+
+接下來我們逐一介紹：
+
+* **異步程式碼**
```
Comment (codingjenny) [inline docs/zh-hant/docs/async.md:64]:
```suggestion
* **非同步程式碼**
```

File: docs/zh-hant/docs/async.md [hunk 5]
```diff
@@ -0,0 +1,443 @@
...
+* **`async` 和 `await`**
+* **協程**
+
+## 異步程式碼
```
Comment (codingjenny) [inline docs/zh-hant/docs/async.md:68]:
```suggestion
## 非同步程式碼
```

File: docs/zh-hant/docs/async.md [hunk 6]
```diff
@@ -0,0 +1,443 @@
...
+
+異步程式碼僅意味著程式語言 💬 有辦法告訴電腦/程式 🤖 在程式碼中的某個點，它 🤖 需要等待某些事情完成。讓我們假設這些事情被稱為 "慢速檔案"📝。
```
Comment (codingjenny) [inline docs/zh-hant/docs/async.md:70]:
```suggestion
非同步程式碼僅意味著程式語言 💬 有辦法告訴電腦/程式 🤖 在程式碼中的某個點，它 🤖 需要等待某些事情完成。讓我們假設這些事情被稱為「慢速檔案」📝。
```

File: docs/zh-hant/docs/async.md [hunk 7]
```diff
@@ -0,0 +1,443 @@
...
+
+因此，在等待 "慢速檔案"📝 完成的這段時間，電腦可以去處理一些其他工作。
```
Comment (codingjenny) [inline docs/zh-hant/docs/async.md:72]:
```suggestion
因此，在等待「慢速檔案」📝 完成的這段時間，電腦可以去處理一些其他工作。
```

File: docs/zh-hant/docs/async.md [hunk 8]
```diff
@@ -0,0 +1,443 @@
...
+
+接著程式 🤖 會在有空檔時回來查看是否有等待的工作已經完成，並執行必要的後續操作。
+
+接下來，它 🤖 完成第一個工作（例如我們的 "慢速檔案"📝）並繼續執行相關的所有操作。
+
+這個 "等待其他事情" 通常指的是一些相對較慢的（與處理器和 RAM 記憶體的速度相比）的 <abbr title="Input and Output">I/O</abbr> 操作，比如說：
```
Comment (codingjenny) [inline docs/zh-hant/docs/async.md:78]:
```suggestion
接下來，它 🤖 完成第一個工作（例如我們的「慢速檔案」📝）並繼續執行相關的所有操作。

這個「等待其他事情」通常指的是一些相對較慢的（與處理器和 RAM 記憶體的速度相比）的 <abbr title="Input and Output">I/O</abbr> 操作，比如說：
```

File: docs/zh-hant/docs/async.md [hunk 9]
```diff
@@ -0,0 +1,443 @@
...
+
+* 通過網路傳送來自客戶端的資料
+* 從網路接收來自客戶端的數據
+* 從磁碟讀取檔案內容
+* 將數據寫入磁碟
+* 遠端 API 調用
```
Comment (codingjenny) [inline docs/zh-hant/docs/async.md:84]:
```suggestion
* 通過網路傳送來自用戶端的資料
* 從網路接收來自用戶端的資料
* 從磁碟讀取檔案內容
* 將內容寫入磁碟
* 遠端 API 操作
```

File: docs/zh-hant/docs/async.md [hunk 10]
```diff
@@ -0,0 +1,443 @@
...
+* 資料庫操作
+* 資料庫查詢
+* 等等
+
+由於大部分的執行時間都消耗在等待 <abbr title="輸入與輸出">I/O</abbr> 操作上，因此這些操作被稱為 "I/O 密集型" 操作。
+
+之所以稱為 "異步"，是因為電腦/程式不需要與那些耗時的任務 "同步"，等待任務完成的精確時間，然後才能取得結果並繼續工作。
```
Comment (codingjenny) [inline docs/zh-hant/docs/async.md:91]:
```suggestion
之所以稱為「非同步」，是因為電腦/程式不需要與那些耗時的任務「同步」，等待任務完成的精確時間，然後才能取得結果並繼續工作。
```

File: docs/zh-hant/docs/async.md [hunk 11]
```diff
@@ -0,0 +1,443 @@
...
+
+相反地，異步系統在任務完成後，可以讓任務稍微等一下（幾微秒），等待電腦/程式完成手頭上的其他工作，然後再回來取得結果繼續進行。
+
+相對於 "異步"（asynchronous），"同步"（synchronous）也常被稱作 "順序性"（sequential），因為電腦/程式會依序執行所有步驟，即便這些步驟涉及等待，才會切換到其他任務。
```
Comment (codingjenny) [inline docs/zh-hant/docs/async.md:95]:
```suggestion
相反地，非同步系統在任務完成後，可以讓任務稍微等一下（幾微秒），等待電腦/程式完成手頭上的其他工作，然後再回來取得結果繼續進行。

相對於「非同步」（asynchronous），「同步」（synchronous）也常被稱作「順序性」（sequential），因為電腦/程式會依序執行所有步驟，即便這些步驟涉及等待，才會切換到其他任務。
```

File: docs/zh-hant/docs/async.md [hunk 12]
```diff
@@ -0,0 +1,443 @@
...
+
+### 並行與漢堡
+
+上述異步程式碼的概念有時也被稱為 "並行"，它不同於 "平行"。
```
Comment (codingjenny) [inline docs/zh-hant/docs/async.md:99]:
```suggestion
上述非同步程式碼的概念有時也被稱為「並行」，它不同於「平行」。
```

File: docs/zh-hant/docs/async.md [hunk 13]
```diff
@@ -0,0 +1,443 @@
+# 並行 async / await
+
+有關路徑操作函數的 async def 語法的細節與異步(asynchronous)程式碼、並行(concurrency)與平行(parallelism)的一些背景知識。
```
[the remark this answers is not in the document]
Comment (ILoveSorasakiHina) [reply docs/zh-hant/docs/async.md]: 感謝提醒，我腦子一直跑出激活函數，下意識就翻成函數，仔細想想激活函數確實是數學的函數而非程式的函式

File: docs/zh-hant/docs/async.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (mattwang44) [reply docs/zh-hant/docs/async.md]: update 一下，最近這件事被改掉囉，中文字之間不需空格
https://github.com/fastapi/fastapi/pull/12509

File: docs/zh-hant/docs/async.md [hunk 14]
```diff
@@ -0,0 +1,442 @@
...
+
+這也是大多數 Web 應用的情況。
+
+許多用戶正在使用你的應用程式，而你的伺服器則在等待 🕙 這些用戶不那麼穩定的網路來傳送請求。
+
+接著，再次等待 🕙 回應。
+
+這種「等待」 🕙 通常以微秒來衡量，但累加起來，最終還是花費了很多等待時間。
+
+這就是為什麼對於 Web API 來說，使用異步程式碼 ⏸🔀⏯ 是非常有意義的。
```
Comment (mattwang44) [inline docs/zh-hant/docs/async.md:248]:
```suggestion
這就是為什麼對於 Web API 來說，使用非同步程式碼 ⏸🔀⏯ 是非常有意義的。
```

File: docs/zh-hant/docs/async.md [hunk 15]
```diff
@@ -0,0 +1,442 @@
...
+
+這種類型的異步性正是 NodeJS 成功的原因（儘管 NodeJS 不是平行的），這也是 Go 語言作為編程語言的一個強大優勢。
```
Comment (mattwang44) [inline docs/zh-hant/docs/async.md:250]:
```suggestion
這種類型的非同步性正是 NodeJS 成功的原因（儘管 NodeJS 不是平行的），這也是 Go 語言作為程式語言的一個強大優勢。
```

File: docs/zh-hant/docs/async.md [hunk 16]
```diff
@@ -0,0 +1,442 @@
...
+
+*是的，這就是全部的故事。*
+
+---
+
+這裡沒有任何需要等待 🕙 的地方，只需要在房子的多個地方進行大量的工作。
+
+你可以像漢堡的例子那樣輪流進行，先打掃客廳，再打掃廚房，但由於你不需要等待 🕙 任何事情，只需要持續地打掃，輪流並不會影響任何結果。
+
+無論輪流執行與否(並行)，你都需要相同的工時完成任務，同時需要執行相同工作量。
```
Comment (mattwang44) [inline docs/zh-hant/docs/async.md:274]:
```suggestion
無論輪流執行與否（並行），你都需要相同的工時完成任務，同時需要執行相同工作量。
```

File: docs/zh-hant/docs/async.md [hunk 17]
```diff
@@ -0,0 +1,442 @@
...
+由於大多數的執行時間都花在實際的工作上（而不是等待），而電腦中的工作由 <abbr title="Central Processing Unit">CPU</abbr> 完成，因此這些問題被稱為「CPU 密集型」。
+
+---
+
+常見的 CPU 密集型操作範例包括那些需要進行複雜數學計算的任務。
+
+例如：
+
+* **音訊**或**圖像處理**；
+* **電腦視覺**: 一張圖片由數百萬個像素組成，每個像素有3個值/顏色，處理這些像素通常需要同時進行大量計算;
```
Comment (mattwang44) [inline docs/zh-hant/docs/async.md:289]:
```suggestion
* **電腦視覺**：一張圖片由數百萬個像素組成，每個像素有 3 個值/顏色，處理這些像素通常需要同時進行大量計算；
```

File: docs/zh-hant/docs/async.md [hunk 18]
```diff
@@ -0,0 +1,442 @@
...
+* **機器學習**: 通常需要大量的「矩陣」和「向量」運算。想像一個包含數字的巨大電子表格，並所有的數字同時相乘;
+* **深度學習**: 這是機器學習的子領域，同樣適用。只不過這不僅僅是一張數字表格，而是大量的數據集合，並且在很多情況下，你會使用特殊的處理器來構建或使用這些模型。
+
+### 並行 + 平行: Web + 機器學習
+
+使用 **FastAPI**，你可以利用並行的優勢，這在 Web 開發中非常常見（這也是 NodeJS 的最大吸引力）。
+
+但你也可以利用平行與多行程(multiprocessing)（讓多個行程同時運行）的優勢來處理機器學習系統中的 **CPU 密集型** 工作。
```
Comment (mattwang44) [inline docs/zh-hant/docs/async.md:297]:
```suggestion
但你也可以利用平行與多行程 (multiprocessing)（讓多個行程同時運行）的優勢來處理機器學習系統中的 **CPU 密集型**工作。
```

File: docs/zh-hant/docs/async.md [hunk 19]
```diff
@@ -0,0 +1,442 @@
...
+
+這一點，再加上 Python 是 **資料科學**、機器學習，尤其是深度學習的主要語言，讓 **FastAPI** 成為資料科學/機器學習 Web API 和應用程式（以及許多其他應用程式）的絕佳選擇。
+
+想了解如何在生產環境中實現這種平行性，請參見 [Deployment](deployment/index.md){.internal-link target=_blank}。
```
Comment (mattwang44) [inline docs/zh-hant/docs/async.md:301]: `Deployment` 要翻譯？

File: docs/zh-hant/docs/async.md [hunk 20]
```diff
@@ -0,0 +1,442 @@
...
+
+## `async` 和 `await`
+
+現代 Python 版本提供一種非常直觀的方式定義異步程式碼。這使得它看起來就像正常的「順序」程式碼，並在適當的時機「等待」。
```
Comment (mattwang44) [inline docs/zh-hant/docs/async.md:305]:
```suggestion
現代 Python 版本提供一種非常直觀的方式定義非同步程式碼。這使得它看起來就像正常的「順序」程式碼，並在適當的時機「等待」。
```

File: docs/zh-hant/docs/async.md [hunk 21]
```diff
@@ -0,0 +1,442 @@
...
+
+當某個操作需要等待才能返回結果，並且支援這些新的 Python 特性時，你可以像這樣編寫程式碼：
```
Comment (mattwang44) [inline docs/zh-hant/docs/async.md:307]:
```suggestion
當某個操作需要等待才能回傳結果，並且支援這些新的 Python 特性時，你可以像這樣編寫程式碼：
```

File: docs/zh-hant/docs/async.md [hunk 22]
```diff
@@ -0,0 +1,442 @@
...
+
+```Python
+burgers = await get_burgers(2)
+```
+
+這裡的關鍵是 `await`。它告訴 Python 必須等待 ⏸ `get_burgers(2)` 完成它的工作 🕙， 然後將結果存儲在 `burgers` 中。如此，Python 就可以在此期間去處理其他事情 🔀 ⏯ （例如接收另一個請求）。
```
Comment (mattwang44) [inline docs/zh-hant/docs/async.md:313]:
```suggestion
這裡的關鍵是 `await`。它告訴 Python 必須等待 ⏸ `get_burgers(2)` 完成它的工作 🕙， 然後將結果儲存在 `burgers` 中。如此，Python 就可以在此期間去處理其他事情 🔀 ⏯ （例如接收另一個請求）。
```

File: docs/zh-hant/docs/async.md [hunk 23]
```diff
@@ -0,0 +1,442 @@
...
+
+要讓 `await` 運作，它必須位於支持異步功能的函式內。為此，只需使用 `async def` 宣告函式：
```
Comment (mattwang44) [inline docs/zh-hant/docs/async.md:315]:
```suggestion
要讓 `await` 運作，它必須位於支持非同步功能的函式內。為此，只需使用 `async def` 宣告函式：
```

File: docs/zh-hant/docs/async.md [hunk 24]
```diff
@@ -0,0 +1,442 @@
...
+
+但同時，使用 `async def` 定義的函式本身也必須被「等待」。所以，帶有 `async def` 函式只能在其他使用 `async def` 定義的函式內呼叫。
+
+那麼，這就像「先有雞還是先有蛋」的問題，要如何呼叫第一個 `async` 函式呢？
+
+如果你使用 FastAPI，無需擔心這個問題，因為「第一個」函式將是你的路徑操作函式，FastAPI 會知道如何正確處理這個問題。
+
+但如果你想在沒有 FastAPI 的情況下使用 `async` / `await`，你也可以這樣做。
+
+### 編寫自己的異步程式碼
```
Comment (mattwang44) [inline docs/zh-hant/docs/async.md:364]:
```suggestion
### 編寫自己的非同步程式碼
```

File: docs/zh-hant/docs/async.md [hunk 25]
```diff
@@ -0,0 +1,442 @@
...
+
+Starlette （和 **FastAPI**） 是基於 <a href="https://anyio.readthedocs.io/en/stable/" class="external-link" target="_blank">AnyIO</a> 實現的，這使得它們與 Python 標準庫兼容 <a href="https://docs.python.org/3/library/asyncio-task.html" class="external-link" target="_blank">asyncio</a> 和 <a href="https://trio.readthedocs.io/en/stable/" class="external-link" target="_blank">Trio</a>。
```
Comment (mattwang44) [inline docs/zh-hant/docs/async.md:366]:
```suggestion
Starlette（和 **FastAPI**）是基於 <a href="https://anyio.readthedocs.io/en/stable/" class="external-link" target="_blank">AnyIO</a> 實作的，這使得它們與 Python 標準函式庫相容 <a href="https://docs.python.org/3/library/asyncio-task.html" class="external-link" target="_blank">asyncio</a> 和 <a href="https://trio.readthedocs.io/en/stable/" class="external-link" target="_blank">Trio</a>。
```

File: docs/zh-hant/docs/async.md [hunk 26]
```diff
@@ -0,0 +1,442 @@
...
+
+但它使處理異步程式碼變得更加容易。
+
+相同的語法（或幾乎相同的語法）最近也被包含在現代 JavaScript（無論是瀏覽器還是 NodeJS）中。
+
+但在此之前，處理異步程式碼要更加複雜和困難。
+
+在較舊的 Python 版本中，你可能會使用多執行緒或 <a href="https://www.gevent.org/" class="external-link" target="_blank">Gevent</a>。但這些程式碼要更難以理解、調試和思考。
+
+在較舊的 NodeJS / 瀏覽器 JavaScript 中，你會使用「回調」，這可能會導致 <a href="http://callbackhell.com/" class="external-link" target="_blank">回調地獄</a>。
```
Comment (mattwang44) [inline docs/zh-hant/docs/async.md:384]: |lang|call | callback|
|:-:|:-:|:-:|
|繁中|呼叫|回呼|
|簡中|調用|回調|

```suggestion
在較舊的 NodeJS / 瀏覽器 JavaScript 中，你會使用「回呼」，這可能會導致<a href="http://callbackhell.com/" class="external-link" target="_blank">回呼地獄</a>。
```

File: docs/zh-hant/docs/async.md [hunk 27]
```diff
@@ -0,0 +1,442 @@
...
+
+## 協程
+
+**協程**只是 `async def` 函式所返回的非常特殊的事物名稱。Python 知道它是一個類似函式的東西，可以啟動它，並且在某個時刻它會結束，但它也可能在內部暫停 ⏸，只要遇到 `await`。
```
Comment (mattwang44) [inline docs/zh-hant/docs/async.md:388]:
```suggestion
**協程**只是 `async def` 函式所回傳的非常特殊的事物名稱。Python 知道它是一個類似函式的東西，可以啟動它，並且在某個時刻它會結束，但它也可能在內部暫停 ⏸，只要遇到 `await`。
```

File: docs/zh-hant/docs/async.md [hunk 28]
```diff
@@ -0,0 +1,442 @@
...
+
+如果你有相當多的技術背景（例如協程、執行緒、阻塞等），並且對 FastAPI 如何處理 `async def` 與常規 `def` 感到好奇，請繼續閱讀。
+
+///
+
+### 路徑操作函数
+
+當你使用 `def` 而不是 `async def` 宣告*路徑操作函式*時，該函式會在外部的執行緒池（threadpool）中執行，然後等待結果，而不是直接呼叫（因為這樣會阻塞伺服器）。
+
+如果你來自於其他不以這種方式運作的非同步框架，而且你習慣於使用普通的 `def` 定義僅進行簡單計算的路徑操作函式，目的是獲得微小的性能增益（大約100奈秒），請注意，在 FastAPI 中，效果會完全相反。在這些情況下，最好使用 `async def`，除非你的路徑操作函式執行阻塞的 <abbr title="輸入/輸出：磁碟讀寫或網路通訊">I/O</abbr> 的代码。
```
Comment (mattwang44) [inline docs/zh-hant/docs/async.md:418]:
```suggestion
如果你來自於其他不以這種方式運作的非同步框架，而且你習慣於使用普通的 `def` 定義僅進行簡單計算的路徑操作函式，目的是獲得微小的性能增益（大約 100 奈秒），請注意，在 FastAPI 中，效果會完全相反。在這些情況下，最好使用 `async def`，除非你的路徑操作函式執行阻塞的 <abbr title="輸入/輸出：磁碟讀寫或網路通訊">I/O</abbr> 的程式碼。
```

File: docs/zh-hant/docs/async.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (codingjenny) [reply docs/zh-hant/docs/async.md]: 不會哈哈，我一開始也是寫「函數」，是 @mattwang44 之前提點我的，現學現賣哈哈！

File: docs/zh-hant/docs/async.md [hunk 29]
```diff
@@ -0,0 +1,442 @@
...
+
+不過，在這兩種情況下，**FastAPI** [仍然很快](index.md#_11){.internal-link target=_blank}至少與你之前的框架相當（或者更快）。
+
+### 依賴項(Dependencies)
```
Comment (mattwang44) [inline docs/zh-hant/docs/async.md:422]: 翻譯尚未確定
https://github.com/fastapi/fastapi/pull/12441#discussion_r1804535440

File: docs/zh-hant/docs/async.md [hunk 30]
```diff
@@ -0,0 +1,442 @@
...
+
+同樣適用於[依賴項](tutorial/dependencies/index.md){.internal-link target=_blank}。如果依賴項是一個標準的 `def` 函式，而不是 `async def`，那麼它在外部的執行緒池被運行。
+
+### 子依賴項
+
+你可以擁有多個相互依賴的依賴項和[子依賴項](tutorial/dependencies/sub-dependencies.md){.internal-link target=_blank} （作為函式定義的參數），其中一些可能是用 `async def` 宣告，也可能是用 `def` 宣告。它們仍然可以正常運作，用 `def` 定義的那些將會在外部的執行緒中呼叫（來自執行緒池），而不是被「等待」。
+
+### 其他輔助函式
+
+你可以直接呼叫任何使用 `def` 或 `async def` 創建的其他輔助函式，FastAPI 不會影響你呼叫它們的方式。
```
Comment (mattwang44) [inline docs/zh-hant/docs/async.md:432]:
```suggestion
你可以直接呼叫任何使用 `def` 或 `async def` 建立的其他輔助函式，FastAPI 不會影響你呼叫它們的方式。
```

File: docs/zh-hant/docs/async.md [hunk 31]
```diff
@@ -0,0 +1,442 @@
...
+
+這與 FastAPI 為你呼叫*路徑操作函式*和依賴項的邏輯有所不同。
+
+如果你的輔助函式是用 `def` 宣告的，它將會被直接呼叫（按照你在程式碼中撰寫的方式），而不是在執行緒池中。如果該函數是用 `async def` 宣告，那麼你在呼叫時應該使用 `await` 等待其結果。
```
Comment (mattwang44) [inline docs/zh-hant/docs/async.md:436]:
```suggestion
如果你的輔助函式是用 `def` 宣告的，它將會被直接呼叫（按照你在程式碼中撰寫的方式），而不是在執行緒池中。如果該函式是用 `async def` 宣告，那麼你在呼叫時應該使用 `await` 等待其結果。
```

File: docs/zh-hant/docs/async.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (codingjenny) [reply docs/zh-hant/docs/async.md]: 沒錯！再麻煩 @ILoveSorasakiHina 幫忙把有空格的地方都刪除，謝謝！

File: docs/zh-hant/docs/async.md [hunk 32]
```diff
@@ -0,0 +1,442 @@
...
+
+///
+
+---
+
+想像你是故事中的電腦或程式 🤖。
+
+當你排隊時，你在放空😴，等待輪到你，沒有做任何「生產性」的事情。但這沒關係，因為收銀員只是接單（而不是準備食物），所以排隊速度很快。
+
+然後，當輪到你時，你開始做真正「有生產力」的工作，處理菜單，決定你想要什麼，替戀人選擇餐點，付款，確認你給了正確的賬單或信用卡，檢查你是否被正確收費，確認訂單中的項目是否正確等等。
```
Comment (codingjenny) [inline docs/zh-hant/docs/async.md:154]:
```suggestion
然後，當輪到你時，你開始做真正「有生產力」的工作，處理菜單，決定你想要什麼，替戀人選擇餐點，付款，確認你給了正確的帳單或信用卡，檢查你是否被正確收費，確認訂單中的項目是否正確等等。
```

Comment (codingjenny) [reply docs/zh-hant/docs/async.md:154]: 雖然「帳」和「賬」為異體字，但慣用的用法還是「帳單」。

File: docs/zh-hant/docs/async.md [hunk 33]
```diff
@@ -0,0 +1,442 @@
...
+
+這家速食店有 8 個處理器（收銀員/廚師）。而並行漢堡店可能只有 2 個處理器（一位收銀員和一位廚師）。
+
+儘管如此，最終的體驗並不是最理想的。😞
+
+---
+
+這是與漢堡類似的故事。🍔
+
+一個更"現實"的例子，想像一間銀行。
```
Comment (codingjenny) [inline docs/zh-hant/docs/async.md:226]:
```suggestion
一個更「現實」的例子，想像一間銀行。
```

**Comment by ILoveSorasakiHina** [comment]
@mattwang44  @codingjenny  除了有爭議的部分，其他地方皆已更新完畢，感謝兩位的review。如有其他需要調整的地方，請務必再提醒我。

File: docs/zh-hant/docs/async.md (hunk 29 above)
Comment (ILoveSorasakiHina) [reply docs/zh-hant/docs/async.md:422]: 根據討論，維持原翻譯。

File: docs/zh-hant/docs/async.md [hunk 34]
```diff
@@ -0,0 +1,442 @@
...
+
+但在此之前，處理異步程式碼要更加複雜和困難。
+
+在較舊的 Python 版本中，你可能會使用多執行緒或 <a href="https://www.gevent.org/" class="external-link" target="_blank">Gevent</a>。但這些程式碼要更難以理解、調試和思考。
+
+在較舊的 NodeJS / 瀏覽器 JavaScript 中，你會使用「回呼」，這可能會導致<a href="http://callbackhell.com/" class="external-link" target="_blank">回呼地獄</a>。
+
+## 協程
+
+**協程**只是 `async def` 函式所回傳的非常特殊的事物名稱。Python 知道它是一個類似函式的東西，可以啟動它，並且在某個時刻它會結束，但它也可能在內部暫停 ⏸，只要遇到 `await`。
```
Comment (celestywang) [inline docs/zh-hant/docs/async.md:388]: I think you need a space after `**協程**`
image

**Review by celestywang — CHANGES_REQUESTED** [review]

File: docs/zh-hant/docs/async.md (hunk 34 above)
Comment (codingjenny) [reply docs/zh-hant/docs/async.md:388]: The rendering issue is fixed in #12509

**Review by yychanlee — APPROVED** [review]

**Review by celestywang — APPROVED** [review]

File: docs/zh-hant/docs/async.md [hunk 35]
```diff
@@ -0,0 +1,442 @@
+# 並行與 async / await
+函數
```
Comment (codingjenny) [inline docs/zh-hant/docs/async.md:2]:
```suggestion

```

Comment (codingjenny) [reply docs/zh-hant/docs/async.md:2]: Just fix this, and others LGTM.

**Review by mattwang44 — APPROVED** [review]

**Comment by ILoveSorasakiHina** [comment]
hi @alejsdev 

I hope this message finds you well. I recently realized that I accidentally deleted my forked repository, which contained the changes for this PR. I'm not sure if this PR has already been merged or if further adjustments are needed from my side.

If this PR is still under review, I’d be happy to re-submit the changes from a new fork. Please let me know how I can assist in moving this forward.

Apologies for the inconvenience, and thank you for your understanding!

Best regards

ILoveSorasakiHina

**Pull request closed without merging**